src/yaml/aslYamlLanguageService.ts

```diff
@@ -0,0 +1,231 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+import * as prettier from 'prettier'
+import {
+    DocumentLanguageSettings,
+    getLanguageService as getLanguageServiceVscode,
+    JSONDocument,
+    JSONSchema,
+    LanguageService,
+    LanguageServiceParams,
+} from 'vscode-json-languageservice'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import {
+    CompletionList,
+    DocumentSymbol,
+    FormattingOptions,
+    Hover,
+    Position,
+    Range,
+    SymbolInformation,
+    TextEdit
+} from 'vscode-languageserver-types'
+import {
+    parse as parseYAML,
+    SingleYAMLDocument,
+    YAMLDocument,
+} from 'yaml-language-server/out/server/src/languageservice/parser/yamlParser07'
+import { YAMLCompletion } from 'yaml-language-server/out/server/src/languageservice/services/yamlCompletion'
+import { YAMLSchemaService } from 'yaml-language-server/out/server/src/languageservice/services/yamlSchemaService'
+import { matchOffsetToDocument } from 'yaml-language-server/out/server/src/languageservice/utils/arrUtils'
+import doCompleteAsl from '../completion/completeAsl'
+
+export const getLanguageService = function(params: LanguageServiceParams, schema: JSONSchema, aslLanguageService: LanguageService): LanguageService {
+    const builtInParams = {}
+
+    const languageService = getLanguageServiceVscode({
+        ...params,
+        ...builtInParams,
+    })
+
+    const requestServiceMock = async function(uri: string): Promise<string> {
+        return new Promise<string>(c => {
+            c(JSON.stringify(schema))
+        })
+    }
+    const schemaService = new YAMLSchemaService(requestServiceMock, params.workspaceContext)
+    // initialize schema
+    schemaService.registerExternalSchema('yasl', ['*.asl.yaml', '*.asl.yml'], schema)
+    schemaService.getOrAddSchemaHandle('yasl', schema)
+
+    const completer = new YAMLCompletion(schemaService)
+
+    languageService.doValidation = async function(
+        textDocument: TextDocument,
+        jsonDocument: JSONDocument,
+        documentSettings: DocumentLanguageSettings
```

Comment:
`jsonDocument` and `documentSettings` parameters are not used anywhere. We can delete them.

Reply:
Removed unused parameters from methods in this module.